### ISSUE TITLE
[2.0rc3] Fetch module cannot deal with multiline argument

### ISSUE THREAD

**Issue by feliksik** [ansible/ansible#13630]
If I use `>` for multiline strings in the fetch module, this goes fine; according to yaml spec, the newlines are stripped. 

```
- name: save public keys to local system
  fetch: >
    src=/home/{{ssh_user}}/{{ssh_key_file}}.pub
    dest=secret/{{environment_prefix}}/hosts/{{inventory_hostname}}/home/{{ssh_user}}/{{ssh_key_file}}.pub
    fail_on_missing=yes
    flat=yes
```

But with `^` it breaks (the yaml resulting string then contains `\n`). The point is: this worked with Ansible 1.9. I think it was an explicit goal to be backwards compatible, so therefore I report it. 

```
- name: save public keys to local system
  fetch: >
    src=/home/{{ssh_user}}/{{ssh_key_file}}.pub
    dest=secret/{{environment_prefix}}/hosts/{{inventory_hostname}}/home/{{ssh_user}}/{{ssh_key_file}}.pub
    fail_on_missing=yes
    flat=yes
```

outputs: 

```
ERROR! this task 'fetch' has extra params, which is only allowed in the following modules: command, shell, script, include, include_vars, add_host, group_by, set_fact, raw, meta

The error appears to have been in '/home/eric/projects/heater/roles/ssh_host/tasks/main.yml': line 7, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

# - install keys on relevant host
- name: save public keys to local system
  ^ here

```

**Comment by edrozenberg** [comment]
Having the same issue with the unarchive module. Ansible release 2.0.0.2

For example the following task:

```
# Unzip the Postgres PrefPane to the Mac PrefPanes system directory
- name: unzip postgres prefpane to system prefpane dir
  unarchive: >
    src={{ ppfile_result.stdout | regex_replace(' ', '\ ') }}
    dest={{ macprefpane_dir }}
    copy=no
    owner=root
    group=admin
    mode=0644
  become: yes
  become_method: sudo
  when: >
    (ppdir_result.stat.isdir is undefined) or
    (ppver_result.stdout != ppfilever_result.stdout)
```

Causes the following error:

```
TASK [desktop/mac : include] ***************************************************
fatal: [zebra]: FAILED! => {"failed": true, "reason": "ERROR! this task 'unarchive' has extra params, which is only allowed in the following modules: command, shell, script, include, include_vars, add_host, group_by, set_fact, raw, meta\n\nThe error appears to have been in '/Users/eduardr/Developer/Work/Ansible/roles/desktop/mac/tasks/pgsql.yaml': line 158, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n# Unzip the Postgres PrefPane to the Mac PrefPanes system directory\n- name: unzip postgres prefpane to system prefpane dir\n  ^ here\n"}

PLAY RECAP *********************************************************************
zebra                      : ok=57   changed=6    unreachable=0    failed=1   
```

Also noticed the error output from Ansible now contains literal "\n" characters that are not properly interpreted by the Mac terminal, so the error messages display the error text all jumbled together.

**Comment by edrozenberg** [comment]
@jimi-c I'm still having the same problem with ansible 2.0.1.0. 

Not sure why reports of this aren't widespread though, maybe few people are using this multi-line style? 

And don't know why unarchive in particular has this issue - I use the same multi-line style with many other modules with no problem.

**Comment by edrozenberg** [comment]
@jimi-c Some further info after digging a bit more:
- Have had the issue with unarchive and file modules
- Issue workaround is to quote args that have {{ }} parameters. For example this doesn't work:

```
- name: unzip postgres prefpane to system prefpane dir
  unarchive: >
    src={{ ppfile_result.stdout | regex_replace(' ', '\ ') }}
    dest={{ macprefpane_dir }}
```

Adding quotes makes it work:

```
- name: unzip postgres prefpane to system prefpane dir
  unarchive: >
    src="{{ ppfile_result.stdout | regex_replace(' ', '\ ') }}"
    dest="{{ macprefpane_dir }}"
```

**Comment by jimi-c** [comment]
@feliksik it looks like we fixed your issue here already, at least what I'm seeing in devel and stable-2.0.

@edrozenberg the patch above appears to resolve your issue, just waiting for tests to pass to confirm it doesn't introduce any other bugs.

**Comment by jimi-c** [comment]
Merged the above into both devel and stable-2.0, so this will be included with 2.0.2.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Comment by hrstoyanov** [comment]
Thank you. It works now.
